src/typecheck/arith.mli

```diff
@@ -1,4 +1,27 @@
 
+(** AE Integer Arithmetic *)
+module Ae: sig
+
+  module Tff
+      (Type : Tff_intf.S)
+      (Ty : Dolmen.Intf.Ty.Ae_Arith with type t := Type.Ty.t)
+      (T : Dolmen.Intf.Term.Ae_Arith with type t := Type.T.t
+                                        and type ty := Type.Ty.t) : sig
+
+    type _ Type.err +=
+      | Expected_arith_type : Type.Ty.t -> Dolmen.Std.Term.t Type.err
+      (** Error raised when an arithmetic type was expected (i.e. either
+          int or real), but another type was found. *)
+      | Incompatible_Arith_Types : Type.Ty.t * Type.Ty.t  -> Dolmen.Std.Term.t Type.err
+      (** Raised when a binary application that expects two values of the same
+          arithmetic type receives two values of different arithmetic types. *)
```

Comment:
Actually, this error could go away, and the code simplified (see my other comment).

Reply:
Done.